app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -84,9 +84,19 @@ def link_identity_to_service_provider
         current_user: current_user,
         ial: resolved_authn_context_int_ial,
         rails_session_id: session.id,
+        email_address_id: email_address_id,
       )
     end
 
+    def email_address_id
+      return nil unless IdentityConfig.store.feature_select_email_to_share_enabled
+      return session[:selected_email_id] if session[:selected_email_id].present?
+      identity = current_user.identities.find_by(service_provider: sp_session['issuer'])
+      email_id = identity&.email_address_id
+      return email_id if email_id.is_a? Integer
+      return EmailContext.new(current_user).last_sign_in_email_address.id
```

Comment:
Also, I wonder if we could optimize this to skip updating the value unless we have `session[:selected_email_id]`. It seems wasteful to query for and call update on `email_address_id` every time someone signs in, if we only really expect it to change from `nil` to non-`nil`, and probably only ever happen once.

There's some prior art to this in `link_identity` where we only assign certain values if they're present. Maybe we could do that with `email_address_id` as well, ignoring any `nil` values?

https://github.com/18F/identity-idp/blob/cf000823195acb5f1f7e30fbe8b749bc14bbad9a/app/services/identity_linker.rb#L47-L48

That way, we could update this method to only return the session value if it exists.

Reply:
Good point. I tested locally and it seems to work OK with the query to identity removed.

Reply:
I don't think this would work as expected. If this method returns `nil`, I'd worry `IdentityLinker` will unset a `email_address_id` if one had been set previously.

I haven't run through the steps, but I'd suspect the issue would be: (This would make for a good feature spec, btw)

1. Sign in to partner
2. When prompted for consent, choose a different email
3. In a separate browser or private browsing window, sign in to partner again

My fear is that at that is point, if there is no `selected_email_id` in the session, it'll update the identity to set `email_address_id` to `nil`, even though we had previously explicitly set an email address to share.